src/linuxforhealth/csvtofhir/fhirrs/basic.py

```diff
@@ -0,0 +1,174 @@
+          
+    # create other identifier - tokenized_sid
+    if incoming_data.tokenized_sid != None and incoming_data.tokenized_sid != "":
+        identifier_id = incoming_data.identifierNamePrefix + ".tokenized_sid"
+        identifier = _build_other_identifier(identifier_id, incoming_data.tokenized_sid, incoming_data.baseSystem, incoming_data.identifierNamePrefix)
+        identifierList.append(identifier)
+        
+    # create other identifier - token_encryption_key
+    if incoming_data.token_encryption_key != None and incoming_data.token_encryption_key != "":
+        identifier_id = incoming_data.identifierNamePrefix + ".token_encryption_key"
+        identifier = _build_other_identifier(identifier_id, incoming_data.token_encryption_key, incoming_data.baseSystem, incoming_data.identifierNamePrefix)
+        identifierList.append(identifier)
+        
+    # create other identifier - source_patient_sid
+    if incoming_data.source_patient_sid != None and incoming_data.source_patient_sid != "":
+        identifier_id = incoming_data.identifierNamePrefix + ".source_patient_sid"
+        identifier = _build_other_identifier(identifier_id, incoming_data.tokenized_sid, incoming_data.baseSystem, incoming_data.identifierNamePrefix)
+        identifierList.append(identifier)
+        
+    # create other identifier - explorys_sid
+    if incoming_data.explorys_sid != None and incoming_data.explorys_sid != "":
+        identifier_id = incoming_data.identifierNamePrefix + ".explorys_sid"
+        identifier = _build_other_identifier(identifier_id, incoming_data.explorys_sid, incoming_data.baseSystem,  incoming_data.identifierNamePrefix)
+        identifierList.append(identifier)
+        
+     # Use local copy of resource_meta so changes do not affect other resources
+     # Passing in none for the source record id because it is not applicable here
+    resource_meta_copy = fhir_utils.add_source_record_id_return_meta_copy(
+        None, resource_meta)
+    
+    # create basic fhir resource with an id if patientInternalIdentifier is present
+    fhir_resource: Basic = Basic.construct(
+        id=resource_id,
+        code=code_cc,
+        identifier=identifierList,
+        meta=resource_meta_copy
+    )
+    
+    # Add patient subject link if the patientInternalIdentifier is present
+    if incoming_data.patientInternalIdentifier != None and incoming_data.patientInternalIdentifier != "":
+        patient_reference = fhir_utils.get_resource_reference_from_str(
+            "Patient",
+            incoming_data.patientInternalIdentifier
+        )
+        fhir_resource.subject = patient_reference
+    
+    # Add created date if it's present
+    if incoming_data.created_date != None and incoming_data.created_date  != "":
+        fhir_resource.created = fhir_utils.get_datetime(incoming_data.created_date)
+
+    resources.append(fhir_resource)
+    return resources
+
+
+# Builds a non token identifier
```

Comment:
Will need to make this comment generic - not specific to a given data set.  Use only the parameters in the signature.

Reply:
Let's dicuss when you get back

Reply:
Done